android/app/src/main/java/org/openbot/robot/CameraActivity.java

```diff
@@ -937,9 +954,7 @@ private void setControlMode(ControlMode controlMode) {
   }
 
   private void connectPhoneController() {
-    if (!phoneController.isConnected()) {
```

Comment:
I think it might make sense to keep this check. Or is this no longer needed?

Reply:
I moved the knowledge of whether the controller is connected or not to the controller itself for two reasons:
1. Simplify the client.
2. We now do `soft` connect and `disconnect`, which means we connection is still open to minimize connection/disconnection, but it still appears to the client like the they are connected/disconnected. This way the client code does not change, while eliminating problems with frequent disconnections. It takes some time for the devices to discover and connect to each other. If somebody switches quickly between controllers, this will interrupt the discovery process and will cause  instability.

Reply:
Great.